### PR TITLE
feat: Add unikraft instance create

### PR DESCRIPTION
**Pull request by jedevc**
Please be gentle in review :pray: 

The changes to `instances.go` are somewhat garbage - this whole editing + patching system needs a heavy rework, but I'm going to do it later.

### CODE REVIEW

File: cmd/unikraft/testdata/TestGolden/instances [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	METRO  NAME               STATE    IMAGE  ARGS  MEMORY  VCPUS  SERVICES  FQDN
+	test   test-<INST>  running  nginx  []    128     1                <DOMAIN>.unikraft.internal
+
+$ unikraft instance inspect test-$UNIQ_INST
+
+stdout:
+	metro:        test
+	name:         test-<INST>
+	uuid:         12345678-1234-1234-1234-123456789abc
+	state:        running
```
Comment (nderjung) [inline cmd/unikraft/testdata/TestGolden/instances:46]: There's an edge-case where the state is in `starting` state (which could also be valid).

File: cmd/unikraft/testdata/TestGolden/instances [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	  - fqdn:     <DOMAIN>.unikraft.internal
+	networks:
+	- uuid:       12345678-1234-1234-1234-123456789abc
+	  private-ip: 10.X.X.X
+	  mac:        aa:bb:cc:dd:ee:ff
+
+$ unikraft instance logs test-$UNIQ_INST
+
+stdout:
+	Powered by Unikraft Kiviuq (0.20.0~f5319a0)55.252 gw 10.X.X.X
```
Comment (nderjung) [inline cmd/unikraft/testdata/TestGolden/instances]: Possible to just check if it `Powered by Unikraft` exists? Output will vary over-time with new Unikraft versions.

**Review by nderjung — COMMENTED** [review]
As discussed offline, will merge as-is - just want to get the test to pass at least as this won't change (see inline comments).

File: cmd/unikraft/testdata/TestGolden/instances (hunk 2 above)
Comment (jedevc) [reply cmd/unikraft/testdata/TestGolden/instances]: Just removed the instance logs entirely for now.

File: cmd/unikraft/testdata/TestGolden/instances (hunk 1 above)
Comment (jedevc) [reply cmd/unikraft/testdata/TestGolden/instances:46]: Added a cleaner for this case. This made me think though, we should have another resource command - `wait`. This command adds a filter-like expression, and waits for it to become true.

So you could do:

```
unikraft instance wait <name> --until state==running
```

Then it would poll at regular intervals, and block until the result is true.

Or maybe this would just be a flag to `create` / `edit` / etc instead. Though I do like the idea of the standalone command.

Comment (nderjung) [reply cmd/unikraft/testdata/TestGolden/instances:46]: I like the idea of a standalone command. I've opened a ticket in triage.

**Review by nderjung — APPROVED** [review]
Thanks!

Reviewed-by: Alexander Jung <alex@unikraft.com>
Approved-by: Alexander Jung <alex@unikraft.com>
